### PR TITLE
Add chord/sequence support to mapper tool

### PR DESCRIPTION
**Pull request by iansan5653**
Adds sequence support to the mapper utility, along with an indicator for the timer and a Reset button. The reset button is useful to be able to clear keys since otherwise it will just keep adding presses (unless the user waits 1.5 seconds).

https://github.com/github/hotkey/assets/2294248/b3a3d9f3-d2b3-40b4-b935-e440ef445122

This is powered by a new `SequenceTracker` class which contains the sequence timer to ensure we keep things consistent across the library and the mapper.

In addition, this PR sets up Actions to deploy Pages so that we don't have to import from unpkg for the pages demos. This allows us to not publish the `SequenceTracker` class publicly because we can now import directly from our own JS code. 

> [!IMPORTANT]
> The Actions workflow requires a repo settings change in https://github.com/github/hotkey/settings/pages, which I don't have access to. All we need to do is change the **Source** to Actions.

### CODE REVIEW

File: src/sequence.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+interface SequenceTrackerOptions {
+  onReset?: () => void
+}
+
+export default class SequenceTracker {
+  static readonly CHORD_TIMEOUT = 1500
+
+  private _path: readonly string[] = []
```
Comment (theinterned) [inline src/sequence.ts:8]: A bit nit, but the underscore seems superfluous and inconsistent with the other private fields.

**Review by theinterned — APPROVED** [review]
There’s one nit, but besides looks great. 

I actually took a bit of a crack at seeing if we could do this without the sequence tracker, and I agree it seems necessary

File: .github/workflows/publish-pages.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      - run: npm install
+
+      - run: npm run build
+
+      - name: Copy files
+        run: |
+          mkdir _site
+          cp -r pages _site
+          cp -r dist _site
+
```
[the remark this answers is not in the document]
Comment (iansan5653) [reply .github/workflows/publish-pages.yml:27]: oh yeah that would be useful. I've been running `npm run build` and then serving with the VSCode live preview, which is a bit cumbersome. Separate PR?

File: src/sequence.ts (hunk 1 above)
Comment (iansan5653) [reply src/sequence.ts:8]: I needed some way to distinguish this from the public `path` getter. Would you prefer a different naming approach? 

I did try using real private class fields (`#path`) here but ran into some compilation issues due to using an old TS version. I think it wouldn't be hard to resolve but I didn't want to get too into the weeds in this PR

Comment (theinterned) [reply src/sequence.ts:8]: Ah didn't realize that we didn't have proper `#private` fields. Carry on then! This makes sense

**Pull request merged**